torch/csrc/autograd/autograd.h

```diff
@@ -1,17 +1,25 @@
-#ifndef THP_AUTOGRAD_H
-#define THP_AUTOGRAD_H
+#pragma once
 
-PyObject * THPAutograd_initExtension(PyObject *_unused);
-void THPAutograd_initFunctions();
+#include <torch/csrc/autograd/variable.h>
 
-namespace torch { namespace autograd {
+#include <ATen/ATen.h>
 
-PyMethodDef* python_functions();
+namespace torch {
+namespace autograd {
+  /// Computes the gradient of current tensor w.r.t. graph leaves.
+  TORCH_API void backward(
+      const variable_list& tensors,
+      const variable_list& grad_tensors={},
+      c10::optional<bool> keep_graph=c10::nullopt,
```

Comment:
nit: would be great to name this `retain_graph` to match the Python API.

Reply:
I noticed that in `variable.h` the C++ API is using `keep_graph` instead of `retain_graph` [here](https://github.com/pytorch/pytorch/blob/e7a9b0d62f6ca6f3e9f350e827ff7b58a2cb7bdb/torch/csrc/autograd/variable.h#L236), Should we also change this API to make these two aligned?

Reply:
hmm yes I think we should (this can happen in a separate PR if you prefer).